sdcm/nemesis.py

```diff
@@ -2161,6 +2188,25 @@ def delete_half_partition(self, ks_cf):
         for pkey, ckey in partitions_for_delete.items():
             queries.append(f"delete from {ks_cf} where pk = {pkey} and ck > {int(ckey[1] / 2)}")
         self.run_deletions(queries=queries, ks_cf=ks_cf)
+        return partitions_for_delete
+
+    def delete_by_range_using_timestamp(self, ks_cf):
+        self.log.debug('Delete by range - using timestamp')
+
+        partitions_for_delete = self.choose_partitions_for_delete(10, ks_cf, with_clustering_key_data=False)
+        if not partitions_for_delete:
+            self.log.error('Not found partitions for delete')
+            return partitions_for_delete
+
+        queries = []
+        partition_percentage = min(max(random.random(), 0.25), 0.75)
+        for pkey, _ in partitions_for_delete.items():
+            self.log.debug("Using USING TIMESTAMP clause in the deletion for this partition: %s", pkey)
+            timestamp = self.get_random_timestamp_from_partition(
+                ks_cf=ks_cf, pkey=pkey, partition_percentage=partition_percentage)
+            queries.append(f"delete from {ks_cf} using timestamp {timestamp} where pk = {pkey}")
```

Comment:
what is expected behavior of such deletion? How it is verified is behaving correctly?

Reply:
yes, this part is missed here

Reply:
Done, I've added some simple checks to see that at least the starting row was affected by deletion (in both mv and base table)